### ISSUE TITLE
ERROR ITMS-90171: "Invalid Bundle Structure"

### ISSUE THREAD

**Issue by cschwarz** [xamarin/xamarin-macios#11728]
### Steps to Reproduce

1. After upgrading to VS 16.10.0 I get the error message below while uploading the IPA file

The error is maybe related to https://github.com/mono/SkiaSharp/issues/1702

### Actual Behavior

```
ERROR ITMS-90171: "Invalid Bundle Structure - The binary file 'App.iOS.app/Frameworks/libSkiaSharp.framework/libSkiaSharp' is not permitted. Your app can’t contain standalone executables or libraries, other than a valid CFBundleExecutable of supported bundles. Refer to the Bundle Programming Guide at https://developer.apple.com/go/?id=bundle-structure for information on the iOS app bundle structure."
```

### Environment

```
Microsoft Visual Studio Community 2019 (2)
Version 16.10.0
VisualStudio.16.Release/16.10.0+31321.278
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

Visual C++ 2019   00435-60000-00000-AA178
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.10.525.31942
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.10.525.31942
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.10.525.31942
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.10.525.31942
Azure Functions and Web Jobs Tools

C# Tools   3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Markdown Editor   1.12.253
A full featured Markdown editor with live preview and syntax highlighting. Supports GitHub flavored Markdown.

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.40423.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.10.15 (552afdf)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.10.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Razor (ASP.NET Core)   16.1.0.2122504+13c05c96ea6bdbe550bd88b0bf6cdddf8cde1725
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   16.0.62105.04180
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.30429.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.10.0-beta.21262.7+1b23bbeda88ea3cb9be9af777f4c99fa8663df81
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

WiX Toolset Visual Studio Extension   1.0.0.4
WiX Toolset Visual Studio Extension version 1.0.0.4
Copyright (c) .NET Foundation and contributors. All rights reserved.

Xamarin   16.10.000.228 (d16-10@1c8354c)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.10.0.115 (remotes/origin/c750fbf1bde3c720d077f51640fe197c6dac7cbe@c750fbf1b)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.10.5 (355b57a)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.3.0.1 (d16-10/22fc2b3)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: b4a3858
    Java.Interop: xamarin/java.interop/d16-10@f39db25
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.35.4@85460d3
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-10@c5732a0


Xamarin.iOS and Xamarin.Mac SDK   14.20.0.1 (fe0e2c518)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Example Project (If Possible)

> VS bug [#1339831](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1339831)

**Comment by spouliot** [comment]
I need to know more about how libSkiaSharp was used (included inside your app) and which version libSkiaSharp was used.

Please attach your build logs (or sample code) to this issue so this can be investigated.

**Comment by cschwarz** [comment]
It is used in a Xamarin.Forms app via SkiaSharp.Views.Forms 2.80.2 nuget package.

I've sent you the build log per mail.

**Comment by spouliot** [comment]
The logs shows SSH error while transferring some files, some `Icon-*.png` files (not all).

I'm not quite sure why. I'm transferring the issue to the team in charge of the XMA agent.

**Comment by spouliot** [comment]
I strongly suspect this is the same issue as https://github.com/xamarin/xamarin-macios/issues/11786

It shows up as "missing icons" for most app - but it also affect other files, as your user framework, could be affected.

Could you try the workarounds ? In particular trying this from Visual Studio for Mac (d16-10, just released yesterday).
Thanks!

**Comment by cschwarz** [comment]
16.10.2 fixed the app icon issues, but I still get the Invalid Bundle Structure error.

**Comment by emaf** [comment]
@spouliot can this be related to the problem you fixed about frameworks not being completly copied from Windows to Mac?

**Comment by spouliot** [comment]
> can this be related to the problem you fixed about frameworks not being completly copied from Windows to Mac?

Yes, recent versions of SkiaSharp were being shipped as a framework.

> '/Users//Library/Caches/Xamarin/mtbs/builds/TestiOSBuildApp.iOS/8c512dc4f4826b88a51c8798b7f65927dc2afe3d8cd845a563df3fddd5f3932e/bin/iPhone/Debug/TestiOSBuildApp.iOS.app/Frameworks/libSkiaSharp.framework/_CodeSignature'.

^ and that can be seen in original linked issue

**Comment by Magendanz** [comment]
I get the following in my build log, but I believe it's the same issue.

`PackageInspectionFailed: Failed to load Info.plist from bundle at path /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.iyFP2M/extracted/TRF.iOS.app/Frameworks/libSkiaSharp.framework; Extra info about "/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.iyFP2M/extracted/TRF.iOS.app/Frameworks/libSkiaSharp.framework/Info.plist": Couldn't stat /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.iyFP2M/extracted/TRF.iOS.app/Frameworks/libSkiaSharp.framework/Info.plist: No such file or directory`

**Comment by cschwarz** [comment]
Fixed with 16.10.3.

**Comment by Magendanz** [comment]
Yes, I can confirm that yesterday's v16.10.3 release of Visual Studio 2019 (with Xamarin.iOS 14.20.24) has also fixed this issue for me.
